src/main/scala/cromwell/binding/values/WdlArray.scala

```diff
@@ -10,5 +10,13 @@ case class WdlArray(wdlType: WdlArrayType, value: Seq[WdlValue]) extends WdlValu
   if (typesUsedInValue.size > 1) {
     throw new UnsupportedOperationException(s"Cannot construct array with a mixed types: $value")
   }
-  override def toWdlString(): String = s"[${value.map{x=>x.toWdlString}.mkString(", ")}]"
+
+  override def toWdlString(): String = s"[${value.map { x => x.toWdlString }.mkString(", ")}]"
+
+  def map[R <: WdlValue](f: WdlValue => R): WdlArray = {
+    value.map{f} match {
+      case s:Seq[R] if s.nonEmpty => WdlArray(WdlArrayType(s.head.wdlType), s)
+      case _ => this
```

Comment:
Style preference:

```scala
case s@(head :: tail) => WdlArray(WdlArrayType(head.wdlType), s)
case Nil => this
```

The first line, I don't know why, but I don't like seeing the extra `:Seq[R]` in there. I'd prefer if just the next line pointed it out with an explicit `Nil` as the only other expected value.

My preference for `(head::tail)` vs `if s.nonEmpty` comes from seeing it in MOOC lectures. Don't know what's best in actual practice.

Happy to defer to the original author / third reviewer overriding though.